vshard/replicaset.lua

```diff
@@ -421,6 +525,26 @@ local function replica_call(replica, func, args, opts)
     assert(opts and opts.timeout)
     replica.activity_ts = fiber_clock()
     local conn = replica.conn
+    if opts.is_async then
+        -- Async call cannot yield. So, we cannot wait for the connection
+        -- to be established and validate vconnect. Immediately fail, if
+        -- something is wrong.
+        local ok, err = conn_validate_vconnect(conn)
+        if not ok then
+            return false, nil, lerror.make(err)
+        end
+    else
+        local timeout, err = conn_wait_vconnect(conn, opts.timeout)
+        if not timeout then
+            return false, nil, lerror.make(err)
+        end
+        local ok
+        ok, err = conn_validate_vconnect(conn)
+        if not ok then
+            return false, nil, lerror.make(err)
+        end
```

Comment:
This block can be done outside of the big `if`. It is the same both for async and sync requests.

Reply:
You're right, done